syntax="proto3";
package eos.console;

message QoSProto {
  message IdentifierProto {
    oneof Identifier {
      uint64 FileId = 1;
      string Path = 2;
    }
  }

  message KVPairProto {
    string key = 1;
    string value = 2;
  }

  message GetProto {
    IdentifierProto identifier = 1;
    repeated string key = 2;
  }

  message SetProto {
    IdentifierProto identifier = 1;
    repeated KVPairProto pair = 2;
  }

  oneof subcmd {
    GetProto get = 1;
    SetProto set = 2;
  }
}
